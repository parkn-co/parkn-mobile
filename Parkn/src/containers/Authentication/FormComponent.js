// @flow

import React from 'react';
import {View, Text, TouchableOpacity, StyleSheet} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {bluePalette, grayPalette} from '../../styles/colors';
import {fontNames} from '../../styles/fonts';
import UnderlinedTextInput from '../../components/UnderlinedInput';

type Props = {
  fields: Array<Object>,
  handleSubmit: Function,
  handleBackPress: Function,
  backButton: any,
  submitButton: any,
};

const FormComponent = ({
  fields,
  handleSubmit,
  handleBackPress,
  backButton,
  submitButton,
}: Props): React.Element<*> => (
  <View style={styles.container}>
    <View style={styles.buttonWrapper}>
      {backButton ? (
        <TouchableOpacity
          onPress={handleBackPress}
          style={styles.navButton}
        >
          <Icon name={backButton.icon} size={30} color="rgba(255,255,255,0.75)" />
          <Text
            style={styles.navText}>
            {backButton.text}
          </Text>
        </TouchableOpacity>
      ) : (
        <View />
      )}

      <TouchableOpacity
        onPress={handleSubmit}
        style={styles.navButton}
      >
        <Text
          style={styles.navText}>
          {submitButton.text}
        </Text>
        <Icon name={submitButton.icon} size={30} color="rgba(255,255,255,0.75)" />
      </TouchableOpacity>
    </View>
    <View style={styles.inputs}>
      {fields.map((field, i) => <UnderlinedTextInput {...field} key={i} />)}
    </View>
  </View>
);

export default FormComponent;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 10,
    backgroundColor: bluePalette.medium,
    justifyContent: 'center',
  },
  navButton: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  navText: {
    color: grayPalette.white,
    opacity: 0.75,
    fontFamily: fontNames.light,
    fontSize: 20,
    marginRight: 5,
    marginLeft: 5,
  },
  inputs: {
    justifyContent: 'center',
  },
  buttonWrapper: {
    position: 'absolute',
    top: 30,
    left: 10,
    right: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
})
