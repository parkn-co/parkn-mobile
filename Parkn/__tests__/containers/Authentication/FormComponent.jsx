/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';

import FormComponent from '../../../src/containers/Authentication/FormComponent';

describe('FormComponent', () => {
  it('renders correctly using snapshots', () => {
    const submitButton = {
      text: 'submit',
      icon: 'satellite',
      onPress: jest.fn(),
    };

    const backButton = {
      text: 'back',
      icon: 'satellite',
      onPress: jest.fn(),
    };

    const fields = [
      {
        value: 'firstname',
        onChangeText: jest.fn(),
      },
    ];

    expect(renderer.create(
      <FormComponent
        fields={fields}
        handleBackPress={jest.fn()}
        submitButton={submitButton}
        backButton={backButton} />
    )).toMatchSnapshot();
  });
});
