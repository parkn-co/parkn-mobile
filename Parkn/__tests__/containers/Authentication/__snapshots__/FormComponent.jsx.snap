exports[`FormComponent renders correctly using snapshots 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#076FEB",
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "left": 10,
        "position": "absolute",
        "right": 10,
        "top": 30,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgba(255,255,255,0.75)",
              "fontFamily": "Material Icons",
              "fontSize": 30,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }>
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "OpenSans-Light",
            "fontSize": 20,
            "marginLeft": 5,
            "marginRight": 5,
            "opacity": 0.75,
          }
        }>
        back
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "OpenSans-Light",
            "fontSize": 20,
            "marginLeft": 5,
            "marginRight": 5,
            "opacity": 0.75,
          }
        }>
        submit
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgba(255,255,255,0.75)",
              "fontFamily": "Material Icons",
              "fontSize": 30,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }>
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "justifyContent": "center",
      }
    }>
    <View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "rgba(255,255,255,0.5)",
              "borderBottomWidth": 1,
              "height": 95,
              "marginBottom": 38,
            },
            Object {},
            Object {},
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "OpenSans-Light",
              "fontSize": 20,
              "opacity": 0.75,
            }
          } />
        <TextInput
          blurOnSubmit={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          selectionColor="#FFFFFF"
          style={
            Object {
              "alignItems": "flex-end",
              "color": "#FFFFFF",
              "flex": 1,
              "fontFamily": "OpenSans-Light",
              "fontSize": 25,
              "paddingBottom": 0,
            }
          }
          value="firstname" />
      </View>
    </View>
  </View>
</View>
`;
